src/gsumm.c

```diff
@@ -1049,6 +1083,17 @@ SEXP glast(SEXP x) {
       if (isunsorted) k = oo[k]-1;
       k = (irowslen == -1) ? k : irows[k]-1;
       dans[i] = dx[k];
+      if (narm){
+        j = 0;
+        while(j < grpsize[i]){
+          const Rcomplex elem = dx[k-j];
+          if (!(ISNAN(elem.i) && ISNAN(elem.r))){
```

Comment:
i don't think we need to do an allocation each iteration, just `ISNAN(dk[k-j].r)`. Actually in this case, I would also use the `||` condition, since it's typically the case that both real & imaginary parts are simultaneously `NA`. So using `||` would short circuit and `dk[k-j].i` wouldn't even need to be run.

Reply:
alternatively, you could flip the condition:

```
if (ISNAN(.) || ISNAN(.)) { j++; next; }
dans[i] = dx[k-j];
break;
```

Reply:
Have to agree flipping the conditional makes it look cleaner